### DISCUSSION TITLE
Graphite as a web component? Interop with a Python web component.

### DISCUSSION THREAD

**Discussion by ToolmakerSteve** [GraphiteEditor/Graphite#1838] [Help]
---------------------
**BACKGROUND**

I see that Graphite is written in Rust. While I am interested in Rust long-term, in the near-term, I am programming in Python or C#.

I have been looking at C# libraries for vector graphics.

Stumbled on Graphite. The direction looks very interesting - I've (literally) dreamed about blending vector and raster graphics off and on over many years (but I went in a different career direction).

My interest is in using (Python or C#) as a scripting language to control an app customized for various purposes.

My code will be MIT licensed.

------------------------
**GOAL**

* Wrap Graphite into a web component.
* Have bi-directional API, so another web component can provide "scripts", that can programmatically control Graphite.

------------------------
**DETAILS**
* In my case, the other webcomponent would be written in Python, or C# Blazor.
* As an analogy, think of writing VBA macros to manipulate an Excel spreadsheet.
  (Which is, believe it or not, the prototype for my first experiment. Written by someone else. Uses shapes and cells of a spreadsheet to represent objects positioned on/moving around a game map. The macros don't run on Web Excel, and my friends don't want to pay for Office 365, just for a game tool. Time to come up with a better approach.)

Searching, I see https://crates.io/crates/webcomponent, which sounds like a starting point for making a Rust Web Component.

But I've never used Rust nor Graphite, so I'm unclear what would be a good "Service API" for Graphite.

----------------
**AN ASIDE**

I've recently started pondering how app development can be less "coding-centric". As much as possible, developed by someone who isn't a coder, then either use a little coding, or call in someone who can code, to add what is hard to do without code. IMPORTANTLY, want non-coders (or "light coders") to be able to maintain and customize the apps. Not be dependent on coders, except when necessary.

**Comment by Keavon** [comment]
Hi, welcome. I'm glad you like our vision.

I don't think this would be that easy to do, although you're welcome to try. Note that Graphite development is moving fast with many refactors changing systems that will be quite an effort to maintain compatibility with. We can't maintain any code for your use case so you'd need to keep rebasing it. Our focus is on making Graphite an all-in-one data processing system for graphical content, which might overlap with your needs more than you expect. Perhaps you can adapt your requirements a bit to use Graphite directly instead of trying to wrap it and maintain a separate connecting API. Good luck!

**Comment by ToolmakerSteve** [comment]
Thanks for replying. I understand. I'll keep an eye on what you are doing.
